zbench.zig

```diff
@@ -152,6 +152,43 @@ pub const Results = struct {
         for (self.results) |r| try r.prettyPrint(writer, colors);
     }
 
+    /// Prints a summary output at the end of benchmarking sessions
+    /// This summary highlights the fastest benchmark by name in green, and compares each
+    /// subsequent benchmark to show how many times slower they are relative to the fastest.
+    pub fn printSummary(self: Results, writer: anytype) !void {
+        if (self.results.len == 0) return;
+
+        const bold_code = Color.bold.code();
+        const reset_code = Color.reset.code();
+        const green_code = Color.green.code();
+        const blue_code = Color.blue.code();
+
+        // Find the fastest result without sorting
+        var fastest_index: usize = 0;
+        var fastest_time = self.results[0].statistics.mean_ns;
+        for (self.results, 0..) |res, i| {
+            if (res.statistics.mean_ns < fastest_time) {
+                fastest_time = res.statistics.mean_ns;
+                fastest_index = i;
+            }
+        }
+
+        // Print Summary Heading in Bold
+        try writer.print("\n \n{s}Summary{s}\n", .{ bold_code, reset_code });
+
+        try writer.print("{s}", .{green_code});
+        try writer.print("{s}", .{self.results[fastest_index].name});
+        try writer.print("{s} ran\n", .{reset_code});
+
+        for (self.results, 0..) |res, i| {
+            if (i == fastest_index) continue; // Skip the fastest since it's already printed
+            const times = @as(f64, @floatFromInt(res.statistics.mean_ns)) / @as(f64, @floatFromInt(fastest_time));
+
+            try writer.print(" └─ {s}{d:.2}x times{s} faster than {s}{s}\n", .{ green_code, times, reset_code, blue_code, res.name });
```

Comment:
It might be a little prettier to print out a line with a `├` character if `i < self.results.len - 1`? Could put that in the formatting parameters to avoid duplicating the whole line.